fabric-renderer-indigo/src/client/java/net/fabricmc/fabric/impl/client/indigo/renderer/render/ItemRenderContext.java

```diff
@@ -317,40 +288,33 @@ public void accept(BakedModel model) {
 		@Override
 		public void accept(BakedModel model, @Nullable BlockState state) {
 			if (hasTransform()) {
```

Comment:
Note: once we move away from the fallback consumer in the FBM default implementation, won't we be routing everything through FRAPI? Isn't that inefficiencient?

Reply:
If it goes through the context's emitter, then yes. However, if the renderer is able to override the default `emitItemQuads` method (as planned), then it can cast the context or emitter to some internal type and then use intrinsics to render the quads and model, making it as efficient as it is now.

I am not sure that the vanilla path is really that fast though; it allocates quite a few different objects while Indigo is able to avoid all allocations. The emissive check and per-quad vertex consumers could provide some slight slowdown, but again this can be avoided using the solution described earlier.

Reply:
Are all renderers expected to override emitBlockQuads / emitItemQuads to provide a faster path? or just Indigo?

The idea was that only Indigo would override `emitBlockQuads`, and just to capture whether a quad is vanilla or not.

Reply:
My impression was that the default implementations would route it only through the quad emitter as exposed through FRAPI, but renderers can override that behavior for either `emit` method to provide extra functionality (like vanilla lighting) or efficiency (like with some vanilla baked models).